src/event_parser.py

```diff
@@ -0,0 +1,86 @@
+from datetime import datetime
+from event_group_rule import EventGroupRule
+from log_event_group import LogEventGroup
+from log_parser import LogEvent, LogParser
+
+""" Module for parsing Log Events into Event Groups """
+
+class LogEventParser:
+    """ Parser for applying group rules against a list of events to
+        create instances of event groups
+        
+        Attributes
+        ----------
+        log_events: list of LogEvent objects to be parsed
+        event_group_rules: list of EventGroupRule objects to be applied
+                           for event grouping
+        event_groups_found: chronological list list of detected 
+                            LogEventGroup objects
+        event_groups_found_dict: dictionary of LogEventGroups where
+                                 key is the group tag """
+
+    def __init__(self, log_events, event_group_rules):
+        self.log_events = log_events
+        self.event_group_rules = event_group_rules
+
+        self.event_groups_found = []
+        self.event_groups_found_dict = {}
+
+    def get_event_search_dict(self, event_group_rule):
+        search_dict = {tag: [] for tag in event_group_rule.trigger_event_tags}
+        return search_dict
+
+    def update_search_dict(self, group_tag, search_dict, event, context_event_buffer):
+        ordered_tags = list(search_dict.keys())
+        event_tag = event.tag
+        tag_index = ordered_tags.index(event_tag)
+        next_tag_index = tag_index + 1
+
+        if tag_index == 0:
```

Comment:
The conditional logic in this function is complex enough that some summary comments explaining what's happening would help.